### PR TITLE
feat: Get and post instrument methods to Empower

### PR DESCRIPTION
**Pull request by SRFU-NN**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,10 +38,7 @@ keyring, or the keyring does not contain the relevant key, you will be prompted
 the password. The password will only be used to get a token from the Empower Web API.
 When the token runs out, you will have to input your password again.
 
-This isn't the best way to use EmpowerHandler, since it is easy to forget to log out,
-which can negatively impact the API server. Therefor, you need to tell that you want to
-use it this way, and you will still get a warning. When you are done developing your
-application, you should only login from a context manager:
+To log in, use the EMpowerHandler with a context manager:
```
Comment (SLCB-NN) [inline README.md:41]: Typo

File: src/OptiHPLCHandler/empower_instrument_method.py [hunk 2]
```diff
@@ -273,7 +275,7 @@ def instrument_method_factory(method_definition: Mapping[str, str]) -> Instrumen
         if method_definition["name"] in ["rAcquityFTN"]:
             logger.debug("Creating SampleManager")
             return SampleManagerMethod(method_definition)
-        elif method_definition["name"] in ["AcquityBSM"]:
+        elif method_definition["name"] in ["AcquityBSM", "ACQ-BSM"]:
```
Comment (SLCB-NN) [inline src/OptiHPLCHandler/empower_instrument_method.py:278]: cant wait to see how long these lists get...

File: src/OptiHPLCHandler/empower_methodset_method.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import logging
-from typing import Union
+from typing import List, Union
 
 from OptiHPLCHandler.data_types import EmpowerMethodSetMethodModel as DataModel
 from OptiHPLCHandler.empower_instrument_method import (
     ColumnOvenMethod,
-    InstrumentMethod,
+    EmpowerInstrumentMethod,
```
Comment (SLCB-NN) [inline src/OptiHPLCHandler/empower_methodset_method.py:7]: I guess you call it EmpowerInstrumentMethod to differentiate from the basis method and the method you will send

File: src/OptiHPLCHandler/empower_methodset_method.py [hunk 4]
```diff
@@ -45,12 +47,19 @@ def __init__(self, method_definition: Union[dict, list]):
                     f"Multiple method set methods found: {method_definition['results']}"
                 )
             method_definition = method_definition["results"][0]
+        self.method_name = method_definition["methodName"]
```
Comment (SLCB-NN) [inline src/OptiHPLCHandler/empower_methodset_method.py:50]: Do we handle if a method has the same name as the method we wish to push?

File: src/OptiHPLCHandler/empower_methodset_method.py [hunk 5]
```diff
@@ -87,3 +97,56 @@ def column_temperature(self, temperature: float):
             raise ValueError("No column oven found in method set method.")
```
Comment (SLCB-NN) [inline src/OptiHPLCHandler/empower_methodset_method.py:97]: Does this include the column oven in the sample handler?

File: tests/test_instrument_method.py [hunk 6]
```diff
@@ -41,17 +41,17 @@ def test_column_handler(self):
     def test_instrument_method(self):
         minimal_definition = {"name": "none_of_the_above"}
         instrument_method = instrument_method_factory(minimal_definition)
-        assert isinstance(instrument_method, InstrumentMethod)
+        assert isinstance(instrument_method, EmpowerInstrumentMethod)
         # This is a PDA, we do not have specific classes for detectors
```
Comment (SLCB-NN) [inline tests/test_instrument_method.py:45]: I have example samplesets for TUV if we wont to create one down the line

File: tests/test_proxy_api.py
Comment (SLCB-NN) [inline tests/test_proxy_api.py:1]: Mocks are still witchcraft to me...

**Review by SLCB-NN — APPROVED** [review]
Just a couple of comments

File: README.md (hunk 1 above)
Comment (SRFU-NN) [reply README.md:41]: Fixed (and added code styling to every mention)

File: src/OptiHPLCHandler/empower_instrument_method.py (hunk 2 above)
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_instrument_method.py:278]: We could to a 
```
if "SBM" in method_definition["name"]
```
instead. That will be broader, but also more risky. Let's discuss.

File: src/OptiHPLCHandler/empower_methodset_method.py (hunk 3 above)
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_methodset_method.py:7]: there is a chance we want to expand it to other chromatographic systems later, and they are bound to work differently, so they would need their own classes.

File: src/OptiHPLCHandler/empower_methodset_method.py (hunk 4 above)
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_methodset_method.py:50]: `EmpowerHandler.PostMethodsetMethod` specifies `?overWriteExisting=false` in the address.

File: src/OptiHPLCHandler/empower_methodset_method.py (hunk 5 above)
Comment (SRFU-NN) [reply src/OptiHPLCHandler/empower_methodset_method.py:97]: Yes, any instrument method which controls a column oven is included. We need to handle them being turned off.

File: tests/test_instrument_method.py (hunk 6 above)
Comment (SRFU-NN) [reply tests/test_instrument_method.py:45]: I think we have one from Waters as well.

File: src/OptiHPLCHandler/empower_methodset_method.py (hunk 4 above)
Comment (SLCB-NN) [reply src/OptiHPLCHandler/empower_methodset_method.py:50]: added to guardrails

**Review by SLCB-NN — APPROVED** [review]

**Pull request merged**